components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -0,0 +1,610 @@
+    private String uuid;
+    private String swagger = null;
+    private JsonNode rootNode;
+    private String requestMethod;
+    private String schemaContent = null;
+
+    public void setLocalentry(String localEntry) {
+        this.uuid = localEntry;
+    }
+
+    /**
+     * Handle the API request message validation.
+     *
+     * @param messageContext Context of the message to be validated
+     * @return Continue true the handler chain
+     */
+    @Override
+    public boolean handleRequest(MessageContext messageContext) {
+        if (logger.isDebugEnabled()) {
+            logger.info("Validating the API request Body content..");
+        }
+        org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext)
+                messageContext).getAxis2MessageContext();
+        String contentType;
+        Object objContentType = axis2MC.getProperty(ThreatProtectorConstants.REST_CONTENT_TYPE);
+        if (objContentType != null) {
+            contentType = objContentType.toString();
+            if (logger.isDebugEnabled()) {
+                logger.info("Content type of the request message :" + contentType);
+            }
+            try {
+                RelayUtils.buildMessage(axis2MC);
+                if (logger.isDebugEnabled()) {
+                    logger.info("Successfully built the request message");
+                }
+                if (ThreatProtectorConstants.APPLICATION_JSON.equals(contentType)) {
+                    try {
+                        initialize(messageContext);
+                    } catch (APIManagementException e) {
+                        logger.error("Error occurred while initializing the swagger elements", e);
+                    }
+                    if (swagger != null) {
+                        requestMethod = messageContext.getProperty(APIMgtGatewayConstants.
+                                ELECTED_REQUEST_METHOD).toString();
+                        if (!APIConstants.SupportedHTTPVerbs.GET.name().equals(requestMethod)) {
+                            try {
+                                validateRequest(messageContext);
+                            } catch (APIManagementException e) {
+                                logger.error("Error occurred while validating the API request", e);
+                            }
+                        }
+                    }
+                }
+            } catch (IOException | XMLStreamException e) {
+                logger.error("Error occurred while building the API request", e);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Validate the response message.
+     *
+     * @param messageContext Context of the API response
+     * @return Continue true the handler chain
+     */
+    @Override
+    public boolean handleResponse(MessageContext messageContext) {
+        if (logger.isDebugEnabled()) {
+            logger.info("Validating the API response  Body content..");
```

Comment:
Please check all debug logs in this file. Seems info is used even if we actually want to print a debug log.